src/Controller/ControllerResolver.php

```diff
@@ -61,41 +53,10 @@ public function getController(Request $request)
 
     /**
      * {@inheritdoc}
+     * @deprecated This method is deprecated. Silex 2.x uses ArgumentResolverInterface for controller arguments
```

Comment:
Why keep this method and implement `ArgumentResolverInterface` on this class? Could it be removed completely?

Reply:
`ControllerResolverInterface` still have this method with comment that it is deprecated a will be removed in 4.0 version so it have to stay until then.

Reply:
I think the implementation should stay too, because we need to support Symfony < 3.1 (because Silex itself supports 2.8+) which does not have a `ArgumentResolverInterface` and still uses this method.

Reply:
I focused on newer version. If you want to support symfony 2.8+ then yes. But then also HttpKernel class have to check that version and depending on it assigning right implementation.

Reply:
Well, as this is a bridge between PHP-DI and Silex, I think it makes sense to support the full dependency range of both libraries (unless there are good reasons not to). And Silex (even in its latest version) supports Symfony 2.8+ (because Symfony 2.8 is an LTS version with support until November 2019).

@mnapoli What's your opinion on this?

Reply:
Yes good point, if it's possible to keep the compatibility let's keep it.

Reply:
Ok, I will check what is needed here and try to add support also for 2.8 version